File: Programmers/pro_43162_네트워크.py
def dfs(cur, visited, computers):
    visited[cur]=1
    for i in range(len(computers)):
        if(computers[cur][i]==1 and visited[i]==0):
            dfs(i,visited,computers)    

def solution(n, computers):
    answer = 0
    visited=[0]*n
    for i in range(n):
        if(visited[i]==0):
            answer+=1
            dfs(i, visited,computers)
    return answer

File: Programmers/pro_43162_네트워크.js

function solution(n, computers) {
  const visited = Array(n).fill(0)
  let answer = 0;
  const dfs = (idx) => {
    visited[idx] = 1;
    computers[idx].forEach((v, i) => {
      if (v === 1 && visited[i] === 0)
        dfs(i)
    })
  }

  for (let i = 0, l = visited.length; i < l; i++) {
    if (!visited[i]) {
      dfs(i)
      answer++;
    }
  }
  return answer;
}